From ce2d485fb030a99217e7536c47737ea7da8da94b Mon Sep 17 00:00:00 2001
From: Priyanka Jain <Priyanka.Jain@freescale.com>
Date: Thu, 9 Jul 2015 15:25:43 +0530
Subject: [PATCH 020/214] dts/ls2085a-rdb: Add support for I2C devices

Update LS2085ARDB device tree to enable support
of below devices which are present on I2C bus:

-RTC DS3232 device
-PCA9547 device
-Thermal Monitor ADT7461 device
-Disable I2C2, I2C3, I2C4 as no device is connected

Signed-off-by: Priyanka Jain <Priyanka.Jain@freescale.com>
[Original patch extracted from FSL LS2085 SDK EAR5.0,
Layerscape2-SDK-SOURCE-20150828-yocto.iso]
Signed-off-by: Jiang Lu <lu.jiang@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-ls2085a-rdb.dts | 32 ++++++++++++++++++++---
 1 file changed, 29 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls2085a-rdb.dts b/arch/arm64/boot/dts/freescale/fsl-ls2085a-rdb.dts
index 8d9bff4..e74b384 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls2085a-rdb.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls2085a-rdb.dts
@@ -51,18 +51,44 @@
 
 &i2c0 {
 	status = "okay";
+	pca9547@75 {
+		compatible = "nxp,pca9547";
+		reg = <0x75>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+		i2c@1 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0x01>;
+			rtc@68 {
+				compatible = "dallas,ds3232";
+				reg = <0x68>;
+			};
+		};
+
+		i2c@3 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0x3>;
+
+			adt7481@4c {
+				compatible = "adi,adt7461";
+				reg = <0x4c>;
+			};
+		};
+	};
 };
 
 &i2c1 {
-	status = "okay";
+	status = "disabled";
 };
 
 &i2c2 {
-	status = "okay";
+	status = "disabled";
 };
 
 &i2c3 {
-	status = "okay";
+	status = "disabled";
 };
 
 &dspi {
-- 
2.5.3

